### PR TITLE
feat: Add DELETE API IBAN [PAGOPA-988] 

### PR DESCRIPTION
**Pull request by FedericoRuzzier**
#### List of Changes
In this pr is described the delete operation for the IBANs, in which, when requested, the IBAN, alongside all the relation with the various attributes and the PA, is removed from the database.

#### Motivation and Context
This addition was required due to the new IBAN implementations. With the new CRUD operations needed for the management of the IBANs, the delete part is crucial for the correct behavior of the system.

#### How Has This Been Tested?
This addition has been tested manually through postman and h2 and through Junit tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/apiconfig/core/controller/IbanController.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+          @ApiResponse(
+              responseCode = "500",
+              description = "Service unavailable",
+              content =
+              @Content(
+                  mediaType = MediaType.APPLICATION_JSON_VALUE,
+                  schema = @Schema(implementation = ProblemJson.class)))
+      })
+  @GetMapping(
+      value = "",
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/apiconfig/core/controller/IbanController.java:98]: where is the GET /ibanV2?

**Comment by FedericoRuzzier** [comment]
update_code

File: src/test/java/it/gov/pagopa/apiconfig/core/service/IbanServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  @Test
+  void deleteIban() throws JSONException, IOException {
+    when(ibanRepository.findByIban(anyString())).thenReturn(Optional.of(getMockIbanEntity()));
+    when(paRepository.findByIdDominio(any())).thenReturn(Optional.of(getMockPa()));
+    when(ibanMasterRepository.findByFkIbanAndFkPa(any(), any())).thenReturn(List.of(getMockIbanMaster()));
+    when(ibanAttributeMasterRepository.findByFkIbanMasterIn(any())).thenReturn(List.of(getMockIbanAttributeMaster()));
+
+    ibanService.deleteIban("1234", "IT99C0222211111000000000000");
+    assertTrue(true);
```
[the remark this answers is not in the document]
Comment (FedericoRuzzier) [reply src/test/java/it/gov/pagopa/apiconfig/core/service/IbanServiceTest.java:463]: ibanService is not mocked but wired, not sure if I can in this case

File: src/main/java/it/gov/pagopa/apiconfig/core/service/IbanService.java [hunk 3]
```diff
@@ -169,6 +169,29 @@ private List<IbanAttributeMaster> saveIbanLabelRelation(IbanV2 iban, IbanMaster
     return labels;
   }
 
+  public String deleteIban(@NotBlank String organizationFiscalCode, @NotNull String ibanValue) {
+    //Get iban entity to be deleted
+    Iban ibanToBeDeleted = getIbanIfExists(ibanValue);
+
+    //Get pa entity
+    Optional<Pa> creditorInstitutionOpt = paRepository.findByIdDominio(organizationFiscalCode);
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/apiconfig/core/service/IbanService.java:177]: export as a method

File: src/main/java/it/gov/pagopa/apiconfig/core/service/IbanService.java [hunk 4]
```diff
@@ -169,6 +169,29 @@ private List<IbanAttributeMaster> saveIbanLabelRelation(IbanV2 iban, IbanMaster
...
+    Pa existingCreditorInstitution = creditorInstitutionOpt.orElseThrow(() -> new AppException(AppError.CREDITOR_INSTITUTION_NOT_FOUND, organizationFiscalCode));
+
+    //Get all ibanMaster relations
+    List<IbanMaster> ibanMastersToBeDeleted = ibanMasterRepository.findByFkIbanAndFkPa(ibanToBeDeleted.getObjId(), existingCreditorInstitution.getObjId());
+
+    //Get all ibanAttributesMaster relations
+    List<IbanAttributeMaster> ibanAttributeMastersToBeDeleted = ibanAttributeMasterRepository.findByFkIbanMasterIn(ibanMastersToBeDeleted.stream().map(e->e.getObjId()).collect(Collectors.toList()));
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/apiconfig/core/service/IbanService.java:184]: split in multiline

**Review by jacopocarlini — APPROVED** [review]
👍

**Pull request merged**